src/py/flwr_example/quickstart_pytorch/mnist.py

```diff
@@ -0,0 +1,434 @@
+    )
+    test_dataset = datasets.MNIST(data_root, train=False, transform=transform)
+
+    # Create partitioned datasets based on the total number of clients and client_id
+    train_loader = dataset_partitioner(
+        dataset=train_dataset,
+        batch_size=train_batch_size,
+        client_id=cid,
+        number_of_clients=nb_clients,
+    )
+
+    test_loader = dataset_partitioner(
+        dataset=test_dataset,
+        batch_size=test_batch_size,
+        client_id=cid,
+        number_of_clients=nb_clients,
+    )
+
+    return (train_loader, test_loader)
+
+
+class MNISTNet(nn.Module):
+    """Simple CNN adapted from Pytorch's 'Basic MNIST Example'."""
+
+    def __init__(self) -> None:
+        super(MNISTNet, self).__init__()
+        self.conv1 = nn.Conv2d(1, 32, 3, 1)
+        self.conv2 = nn.Conv2d(32, 64, 3, 1)
+        self.dropout1 = nn.Dropout2d(0.25)
+        self.dropout2 = nn.Dropout2d(0.5)
+        self.fc1 = nn.Linear(9216, 128)
+        self.fc2 = nn.Linear(128, 10)
+
+    # pylint: disable-msg=arguments-differ,invalid-name
+    def forward(self, x: Tensor) -> Tensor:
+        """Compute forward pass.
+
+        Parameters
+        ----------
+        x: Tensor 
+            Mini-batch of shape (N,28,28) containing images from MNIST dataset.
+            
+
+        Returns
+        -------
+        output: Tensor
+            The probability density of the output being from a specific class given the input.
+
+        """
+        x = self.conv1(x)
+        x = F.relu(x)
+        x = self.conv2(x)
+        x = F.relu(x)
+        x = F.max_pool2d(x, 2)
+        x = self.dropout1(x)
+        x = torch.flatten(x, 1)
+        x = self.fc1(x)
+        x = F.relu(x)
+        x = self.dropout2(x)
+        x = self.fc2(x)
+        output = F.log_softmax(x, dim=1)
+        return output
+
+
+def train(
+    model: torch.nn.ModuleList,
+    train_loader: torch.utils.data.DataLoader,
+    epochs: int,
+    device: torch.device = torch.device("cpu"),
+) -> int:
+    """Train routine based on 'Basic MNIST Example'
+
+    Parameters
+    ----------
+    model: torch.nn.ModuleList
+        Neural network model used in this example.
+        
+    train_loader: torch.utils.data.DataLoader
+        DataLoader used in traning.
+        
+    epochs: int 
+        Number of epochs to run in each round. 
+        
+    device: torch.device 
+         (Default value = torch.device("cpu"))
+         Device where the network will be trained within a client.
+
+    Returns
+    -------
+    num_examples_train: int
+        Number of total samples used during traning.
+
+    """
+    model.train()
+    optimizer = optim.Adadelta(model.parameters(), lr=1.0)
+    scheduler = StepLR(optimizer, step_size=1, gamma=0.7)
+    print(f"Training {epochs} epoch(s) w/ {len(train_loader)} mini-batches each")
+    for epoch in range(epochs):  # loop over the dataset multiple time
+        print()
```

Comment:
Is this empty print necessary?

Reply:
The print in line 250 updates the line. This print is not really necessary but it keeps the terminal pretty. Should I delete it?

Reply:
Ah I see, we can leave it like that